### PR TITLE
Update collector's HPA to use "autoscaling/v2" GV

### PR DESCRIPTION
**Pull request by seh**
The _HorizontalPodAutoscaler_ manifest composed for the OpenTelemetry Collector running in "deployment" mode was using an API group version—"autoscaling/v2beta1"—that was deprecated in Kubernetes version 1.22 and [is no longer served in Kubernetes version 1.25](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#horizontalpodautoscaler-v125). Replace our use of that API group version with the newer "autoscaling/v2," which [has been available at GA quality since Kubernetes version 1.23](https://kubernetes.io/blog/2021/12/07/kubernetes-1-23-release-announcement/#horizontalpodautoscaler-v2-graduates-to-ga).

### CODE REVIEW

File: charts/opentelemetry-demo/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 type: application
 name: opentelemetry-demo
-version: 0.12.7
+version: 0.12.8
```
[the remark this answers is not in the document]
Comment (seh) [reply charts/opentelemetry-demo/Chart.yaml:4]: Do you mean that I should revert all the changes to the "opentelemetry-demo" chart, or that this divergence is acceptable to commit for now?

**Review by TylerHelmuth — COMMENTED** [review]
@seh have you been able to confirm this change locally? We don't have a CI scenario for HorizontalPodAutoscaler.

**Comment by seh** [comment]
> Have you been able to confirm this change locally?

Yes, I did, in a Kubernetes cluster running version 1.24.7 today. It works as expected.

**Comment by seh** [comment]
> Could we also add entry in the https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/UPGRADING.md

Please see my recent commit for such a note.

**Comment by carnei-ro** [comment]
What about using `.Capabilities.KubeVersion.GitVersion` ?

```
{{- if semverCompare ">=1.26-0" .Capabilities.KubeVersion.GitVersion -}}
autoscaling/v2
{{- else if semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion -}}
autoscaling/v2beta2
{{ - else -}}
autoscaling/v2beta1
{{- end }}
```

**Comment by seh** [comment]
Doing so requires a different shape for some of the manifest fields. It’s _possible_, but I don’t think it’s worth the complexity.

**Comment by carnei-ro** [comment]
it is the price to pay to support multiple versions of K8s. Cutting to the `autoscaling/v2` new versions of the chart will only support k8s 1.25+

**Comment by TylerHelmuth** [comment]
My understanding was that `autoscaling/v2` is supported as stable in 1.23.  If that is not correct then we will need changes.  @seh can you confirm this change works in a 1.23 cluster?

**Comment by carnei-ro** [comment]
Yep, my bad, `autoscaling/v2` is supported as stable in 1.23.

**Comment by TylerHelmuth** [comment]
I am ok targeting only 1.23+ support for this chart as I believe that is the minimum version receiving support by the k8s community.  @dmitryax do you agree?

**Pull request merged**